src/Microsoft.IdentityModel.Tokens/Encryption/AuthenticatedEncryptionProvider.cs

```diff
@@ -67,20 +72,120 @@ public AuthenticatedEncryptionProvider(SecurityKey key, string algorithm)
             if (string.IsNullOrWhiteSpace(algorithm))
                 throw LogHelper.LogArgumentNullException(nameof(algorithm));
 
-            _authenticatedkeys = new Lazy<AuthenticatedKeys>(CreateAuthenticatedKeys);
-            _hmacAlgorithm = GetHmacAlgorithm(algorithm);
             Key = key;
             Algorithm = algorithm;
             _cryptoProviderFactory = key.CryptoProviderFactory;
 
+            if (SupportedAlgorithms.IsSupportedEncryptionAlgorithm(algorithm, key))
+            {
+                if (SupportedAlgorithms.IsAesGcm(algorithm))
+                {
+#if NET_CORE
```

Comment:
It would be good to add a test for this exception.
I'd expect the ifdef to be for NETSTANDARD2_0.

Reply:
This has been added. Thanks @GeoK for sharing info on how to test this.